src/main.c

```diff
@@ -7,33 +7,47 @@
 #include "malloc.h"
 #include "asm.h"
 #include "task.h"
+#include "lcd.h"
 
-void hello_world(void)
+void test_lcd(void)
 {
-    char buf[] = "Hello world!\r\n";
-
-    while (1)
-    {
-        uart_write(UART2, buf, 14);
-        yield();
-    }
-}
-
-void dummy(void)
-{
-
-}
-
-void test3(void)
-{
-    char buf[] = "0\r\n";
+    GPIO_pin enable_pin = {
+        .pin = GPIO_PIN_7,
+        .group = GPIOC,
+    };
+    GPIO_pin rs_pin = {
+        .pin = GPIO_PIN_9,
+        .group = GPIOA,
+    };
+    GPIO_pin data_pin_0 = {
+        .pin = GPIO_PIN_5,
+        .group = GPIOB,
+    };
+    GPIO_pin data_pin_1 = {
+        .pin = GPIO_PIN_4,
+        .group = GPIOB,
+    };
+    GPIO_pin data_pin_2 = {
+        .pin = GPIO_PIN_10,
+        .group = GPIOB,
+    };
+    GPIO_pin data_pin_3 = {
+        .pin = GPIO_PIN_8,
+        .group = GPIOA,
+    };
 
-    for (int i = 0; i < 10; i++)
-    {
-        buf[0] = i + '0';
-        uart_write(UART2, buf, 3);
-        yield();
-    }
+    LCD_handler lcd_handler = {
+        .enable_pin = &enable_pin,
```

Comment:
`GPIO_pin` structure is small, so it is better to embed it in the `LCD_handler` structure